packages/web5/lib/src/pexv2/pd.dart

```diff
@@ -0,0 +1,274 @@
+import 'dart:convert';
+import 'dart:math';
+import 'dart:typed_data';
+import 'package:convert/convert.dart';
+import 'package:json_path/json_path.dart';
+import 'package:json_schema/json_schema.dart';
+
+/// PresentationDefinition represents a DIF Presentation Definition defined [here].
+/// Presentation Definitions are objects that articulate what proofs a Verifier requires.
+///
+/// [here]: https://identity.foundation/presentation-exchange/#presentation-definition
+class PresentationDefinition {
+  String id;
+  String? name;
+  String? purpose;
+  List<InputDescriptor> inputDescriptors;
+
+  PresentationDefinition({
+    required this.id,
+    this.name,
+    this.purpose,
+    required this.inputDescriptors,
+  });
+
+  factory PresentationDefinition.fromJson(Map<String, dynamic> json) =>
+      PresentationDefinition(
+        id: json['id'],
+        name: json['name'],
+        purpose: json['purpose'],
+        inputDescriptors: List<InputDescriptor>.from(
+          json['input_descriptors'].map((x) => InputDescriptor.fromJson(x)),
+        ),
+      );
+
+  Map<String, dynamic> toJson() => {
+        'id': id,
+        'name': name,
+        'purpose': purpose,
+        'input_descriptors':
+            List<dynamic>.from(inputDescriptors.map((x) => x.toJson())),
```

Comment:
`inputDescriptors?.map((ind) => ind.toJson()).toList(),`

feels a bit easier to read imo.

Reply:
done